homeassistant/components/google_assistant/helpers.py

```diff
@@ -425,7 +426,7 @@ async def sync_serialize(self, agent_user_id):
                 "webhookId": self.config.local_sdk_webhook_id,
                 "httpPort": self.hass.http.server_port,
                 "httpSSL": self.hass.config.api.use_ssl,
-                "baseUrl": self.hass.config.api.base_url,
+                "baseUrl": async_get_url(self.hass, prefer_external=True),
```

Comment:
I think this is correct. @bramkragten can you verify?

Reply:
This particular change was made Bram, had to ensure I had it correct.


It is not actively used, it is used as a lookup and has to match the base_url in the mDNS announcement (which uses the same logic).

Will be removed soon and replaced by the UUID.